airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -63,7 +64,12 @@
 public class MssqlSource extends AbstractJdbcSource<JDBCType> implements Source {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MssqlSource.class);
-
+  public static final String NULL_CURSOR_VALUE_WITH_SCHEMA_QUERY =
+      """
+        SELECT t1.allowNulls & t2.hasNulls AS %s
+        FROM (SELECT CAST(COLUMNPROPERTY(OBJECT_ID('%s.%s'), '%s', 'AllowsNull') AS BIT) AS allowNulls) AS t1
+        CROSS APPLY (SELECT CAST(IIF(EXISTS(SELECT TOP 1 1 FROM "%s"."%s" WHERE "%s" IS NULL), 1, 0) AS BIT) AS hasNulls) AS t2
```

Comment:
If t1 is `0`, would it still need to run the `SELECT … WHERE … IS NULL`?
I'm not very familiar with sql server's `CROSS APPLY`.
If possible we should avoid the potentially expensive SELECT for NULLs if column is not nullable.

Reply:
The CROSS APPLY operator returns only those rows from the left table expression (in its final output) if it matches with the right table expression. In other words, the right table expression returns rows for the left table expression match only.

Reply:
I see, but do you know if it actually runs the right one (SELECT … IS NULL) if t1 bit is 0?
Would it still look for a match?

Reply:
@rodireich modified query to execute (SELECT … IS NULL) only in case of column is nullable